packages/react-data-grid/src/HeaderCell.tsx

```diff
@@ -1,168 +1,118 @@
-    };
-
-    event.preventDefault();
-    window.addEventListener('mousemove', onMouseMove);
-    window.addEventListener('mouseup', onMouseUp);
-  };
-
-  private onTouchStart = (event: React.TouchEvent) => {
-    const touch = event.changedTouches[0];
-    const { identifier } = touch;
-    const { right } = event.currentTarget.getBoundingClientRect();
-    const offset = right - touch.clientX;
-
-    if (offset > 11) { // +1px to account for the border size
-      return;
-    }
-
-    function getTouch(event: TouchEvent) {
-      for (const touch of event.changedTouches) {
-        if (touch.identifier === identifier) return touch;
-      }
-      return null;
-    }
-
-    const onTouchMove = (event: TouchEvent) => {
-      const touch = getTouch(event);
-      if (touch) {
-        this.onResize(touch.clientX + offset);
-      }
-    };
-
-    const onTouchEnd = (event: TouchEvent) => {
-      const touch = getTouch(event);
-      if (!touch) return;
-      window.removeEventListener('touchmove', onTouchMove);
-      window.removeEventListener('touchend', onTouchEnd);
-    };
-
-    window.addEventListener('touchmove', onTouchMove);
-    window.addEventListener('touchend', onTouchEnd);
-  };
-
-  private onResize(x: number) {
-    const width = this.getWidthFromMouseEvent(x);
-    if (width > 0) {
-      this.props.onResize(this.props.column, width);
-    }
-  }
-
-  private getWidthFromMouseEvent(x: number): number {
-    return x - this.cell.current!.getBoundingClientRect().left;
-  }
-
-  getCell() {
-    const { height, column, rowType, allRowsSelected, onAllRowsSelectionChange } = this.props;
-    const renderer = this.props.renderer || SimpleCellRenderer;
+export default function HeaderCell<R, K extends keyof R>({
+  height,
+  column,
+  allRowsSelected,
+  onAllRowsSelectionChange,
+  scrollLeft,
+  ...props
+}: HeaderCellProps<R, K>) {
+  function getCell() {
+    const renderer = column.headerRenderer || SimpleCellRenderer;
     if (isElement(renderer)) {
+      // TODO: is this needed?
       // if it is a string, it's an HTML element, and column is not a valid property, so only pass height
       if (typeof renderer.type === 'string') {
-        return React.cloneElement(renderer, { height });
+        return cloneElement(renderer, { height });
       }
-      return React.cloneElement(renderer, { column, height });
+      return cloneElement(renderer, { column, height });
     }
-    return React.createElement(renderer, { column, rowType, allRowsSelected, onAllRowsSelectionChange });
+    return createElement(renderer, { column, allRowsSelected, onAllRowsSelectionChange });
   }
 
-  setScrollLeft(scrollLeft: number) {
-    const node = this.cell.current;
-    if (node) {
-      node.style.transform = `translateX(${scrollLeft}px)`;
-    }
-  }
+  let cell = getCell();
 
-  removeScroll() {
-    const node = this.cell.current;
-    if (node) {
-      node.style.transform = 'none';
-    }
+  if (column.sortable) {
```

Comment:
I think composition makes this code cleaner and easier to maintain and can add another `if (draggable)` block in the future. It will be nice to completely drop `react-dnd` dependency

Reply:
I think it's still worth using the `react-dnd` as it provides some features that we don't want to rework. I have a PR to upgrade it by using v10, will assign you for review later.